### PR TITLE
Throughput in msg/s

### PR DESCRIPTION
**Pull request by Scooletz**
Connects to https://github.com/Particular/LaunchWave.DevOpsAdvancedMonitoring/issues/143

This PR introduced Throughput in `messages/s` available for both: endpoints and their instances. 

This is done by:
- introducing a new aggregating function `IntervalsAggregator.AggregateTotalMeasurementsPerSecond`
- applying it on any store that measures per message statistic (`processingTimeStore` has been selected)
- dividing any obtained result by the number of seconds in one interval of the specific `HistoryPeriod` that the report is run for

### CODE REVIEW

**Comment by Scooletz** [comment]
Good point @mikeminutillo As far as I see it might be not reporter if the pipeline blows out:

https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Pipeline/MainPipelineExecutor.cs#L34

My questions is: should I take it into a consideration?

**Comment by WilliamBZA** [comment]
@mikeminutillo Processing time should NOT be reported for failed messages. The processing time metric is how long it took to successfully process the message, excluding retries etc.

**Comment by Scooletz** [comment]
@WilliamBZA Does it make this calculations wrong or right? What's the `throughput`? `The number of successfully processed messages per seconds` or `The number of retrieved messages per seconds` ?

**Comment by WilliamBZA** [comment]
I think that Throughput should also be based on successful messages, so I think we're good here.

**Comment by WilliamBZA** [comment]
By that I mean that I don't think it makes sense to say that an endpoint is processing 300 messages a second, but those are all trapped in FLR+SLR and the actual number of successful messages is more like 5 messages a second.

File: src/ServiceControl.Monitoring/Http/Diagrams/DiagramApiModule.cs [hunk 1]
```diff
@@ -38,18 +39,24 @@ public MonitoredEndpointsModule(EndpointRegistry endpointRegistry, CriticalTimeS
...
                 FillInInstanceData(instances, retriesStore, period, (e, v) => e.Retries = v, IntervalsAggregator.AggregateRetries);
+                FillInInstanceData(instances, processingTimeStore, period, (e, v) => e.Throughput = v, intervals => IntervalsAggregator.AggregateTotalMeasurementsPerSecond(intervals, period));
 
                 return Negotiate.WithModel(instances);
             };
         }
 
-        static void FillInEndpointData(MonitoredEndpoint[] endpoints, 
-            VariableHistoryIntervalStore store, 
+        HistoryPeriod ExtractHistoryPeriod()
```
Comment (WilliamBZA) [inline src/ServiceControl.Monitoring/Http/Diagrams/DiagramApiModule.cs]: ❤️

**Pull request merged**